SD2019StackOverflowApp/src/main/java/ro/utcn/sd/mca/SD2019StackOverflowApp/persistence/jdbc/JdbcRepository.java

```diff
@@ -0,0 +1,85 @@
+package ro.utcn.sd.mca.SD2019StackOverflowApp.persistence.jdbc;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.entity.DatabaseEntity;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.persistence.Repository;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.persistence.Specification;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@RequiredArgsConstructor
+class JdbcRepository<T extends DatabaseEntity> implements Repository<T> {
+    private final JdbcTemplate template;
+
+    @Override
+    public synchronized T save(T databaseEntity) {
```

Comment:
You don't need to have this method synchronized: the JdbcTemplate is thread safe.